labgrid/driver/serialdriver.py

```diff
@@ -38,7 +37,7 @@ class SerialDriver(ConsoleExpectMixin, Driver, ConsoleProtocol):
     def __attrs_post_init__(self):
         super().__attrs_post_init__()
         self.logger = logging.getLogger("{}({})".format(self, self.target))
-        if isinstance(self.port, SerialPort):
+        if isinstance(self.port, target_factory.registered_class_from_string("SerialPort")):
```

Comment:
Please don't replace static instances in normal code (for readability and performance reasons).